tests/unit/client.py

```diff
@@ -245,7 +245,7 @@ def create_setup(request):
     initialize_db()
     proc = run_server(['haas', 'serve', '8888' ])
     import time
-    time.sleep(0.5)
+    time.sleep(1)
```

Comment:
You still have a race condition, even after this change. In theory this will be true no matter what the parameter to sleep is, and a truly "safe" value is probably going to make our ci run longer than we'd like.

There's a fixture in `tests/integration/keystone.py` on #615 that does this correctly. That should be merged relatively soon, maybe when it is you can pull it in, abstract out that logic and use it here as well?